services/cdn/wait/wait_test.go

```diff
@@ -0,0 +1,336 @@
+
+type response struct {
+	state  string
+	status int
+}
+
+type apiClientMock struct {
+	responseIndex int
+	responses     []response
+}
+
+// GetDistributionExecute implements APIClientInterface.
+func (mock *apiClientMock) GetDistributionExecute(_ context.Context, _, _ string) (*cdn.GetDistributionResponse, error) {
+	response := mock.responses[mock.responseIndex]
+	mock.responseIndex++
+	mock.responseIndex %= len(mock.responses)
+	if response.status >= http.StatusBadRequest {
+		return nil, &oapierror.GenericOpenAPIError{
+			StatusCode:   response.status,
+			ErrorMessage: "simulated error",
+		}
+	}
+	return &cdn.GetDistributionResponse{
+		Distribution: &cdn.Distribution{
+			Id:     &testDistributionId,
+			Status: &response.state,
+		},
+	}, nil
+}
+
+var (
+	testProjectId      = "testProjectId"
+	testDistributionId = "testDistributionId"
+)
+
+func TestCreateDistributionPoolWaitHandler(t *testing.T) {
+	type args struct {
+		projectId      string
+		distributionId string
+	}
+	type fields struct {
+		responses []response
+	}
+	tests := []struct {
+		name      string
+		args      args
+		fields    fields
+		wantState string
+		wantErr   bool
+	}{
+		{
+			"success immediately",
+			args{
+				projectId:      testProjectId,
+				distributionId: testDistributionId,
+			},
+			fields{
+				responses: []response{
+					{DistributionStateActive, 0},
+				},
+			},
+			DistributionStateActive,
+			false,
+		},
+		{
+			"success delayed",
+			args{
+				projectId:      testProjectId,
+				distributionId: testDistributionId,
+			},
+			fields{
+				responses: []response{
+					{DistributionStateUpdating, 0},
+					{DistributionStateUpdating, 0},
+					{DistributionStateActive, http.StatusOK},
+				},
+			},
+			DistributionStateActive,
+			false,
+		},
+		{
+			"request fails",
+			args{
+				projectId:      testProjectId,
+				distributionId: testDistributionId,
+			},
+			fields{
+				responses: []response{
+					{DistributionStateUpdating, 0},
+					{DistributionStateUpdating, 0},
+					{DistributionStateUpdating, http.StatusInternalServerError},
+				},
+			},
+			"",
+			true,
+		},
+		{
+			"timeout",
+			args{
+				projectId:      testProjectId,
+				distributionId: testDistributionId,
+			},
+			fields{
+				responses: []response{
+					{DistributionStateUpdating, 0},
+				},
+			},
+			DistributionStateUpdating,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			apiClientMock := apiClientMock{
+				responses: tt.fields.responses,
+			}
+			ctx := context.Background()
+			handler := CreateDistributionPoolWaitHandler(ctx, &apiClientMock, tt.args.projectId, tt.args.distributionId)
+			handler.SetTimeout(2 * time.Second)
+			handler.SetSleepBeforeWait(100 * time.Millisecond)
+			handler.SetThrottle(50 * time.Millisecond)
+			response, err := handler.WaitWithContext(ctx)
+			if (err != nil) != tt.wantErr {
+				t.Fatalf("handler error %s, wantErr %v", err, tt.wantErr)
+			}
```

Comment:
I think a return would make here more sense
```suggestion
			if (err != nil) != tt.wantErr {
				t.Fatalf("handler error %s, wantErr %v", err, tt.wantErr)
				return
			}
```

Reply:
I could add it, but the fatalf will abort the testcase anyways